### PR TITLE
Very simple samples for DefaultAzureCredential

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Truth in advertising.

Disclaimer 1: There was some surprise expressed by others that we'd even be spinning samples for DefaultAzureCredential so I kept this rather terse.  Shout out if more (or less?) detailed comments or further samples (other services) would be desired.

Disclaimer 2: Storage bits have been going into preview5, if we want to keep that distinct I can split this into two (eh vs storage)

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/tests/test_blob_samples_authentication.py [hunk 1]
```diff
@@ -125,3 +125,21 @@ def test_auth_shared_access_signature(self):
...
+        default_credential = DefaultAzureCredential()
+ 
+        # Instantiate a BlobServiceClient using a token credential
+        from azure.storage.blob import BlobServiceClient
+        blob_service_client = BlobServiceClient(account_url=self.oauth_url, credential=default_credential)
+        # [END create_blob_service_client_oauth]
+ 
+        # Get account information for the Blob Service
+        account_info = blob_service_client.get_service_properties()
+        assert account_info is not None
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/tests/test_blob_samples_authentication.py]: 1) remove the assert Statement
2) rename to def auth_* insteard of def test_auth_*

Just wanted to make it consistent with https://github.com/Azure/azure-sdk-for-python/pull/8171

Comment (bryevdv) [reply sdk/storage/azure-storage-blob/tests/test_blob_samples_authentication.py]: Shouldn't things also move from `tests` to `samples` in that case?

Comment (KieranBrantnerMagee) [reply sdk/storage/azure-storage-blob/tests/test_blob_samples_authentication.py]: ++ Yep, will also be doing that @bryevdv

**Comment by KieranBrantnerMagee** [comment]
Splitting this into two PRs so that blob samples can go under new folder structure into distinct branch, and eventhubs samples can go into master.

**Pull request closed without merging**